paper/forestecology.Rmd

```diff
@@ -0,0 +1,456 @@
+
+```{r scbi-focal-vs-comp-map, out.width="66%", echo = FALSE, fig.cap="Identify all focal and corresponding competitor trees: All 20 competitor trees of focal tree 4."}
+comp_radius <- growth_scbi %>% 
+  filter(stemID == 4) %>% 
+  st_buffer(dist = comp_dist)
+
+comp_radius_box <- growth_scbi %>% 
+  filter(stemID == 4) %>% 
+  st_buffer(dist = comp_dist + 10) %>% 
+  st_bbox() %>% 
+  st_as_sfc() %>% 
+  st_intersects(growth_scbi, ., sparse = FALSE)
+
+growth_scbi_subset <- growth_scbi %>% 
+  mutate(inside = comp_radius_box) %>% 
+  filter(inside)
+
+ggplot() +
+  geom_sf(data = growth_scbi_subset, aes(col = sp, size = dbh1)) +
+  geom_sf(data = comp_radius, linetype = "dashed", alpha = 0, inherit.aes = FALSE) +
+  coord_sf(xlim = st_bbox(comp_radius)[c("xmin", "xmax")], ylim = st_bbox(comp_radius)[c("ymin", "ymax")]) +
+  geom_sf_text(data = growth_scbi %>% filter(stemID == 4), aes(label = stemID), size = 12) +
+  labs(size = "DBH at first census", col = "species") 
+```
+
+Using the `unnest()` function from the `tidyr` package, we can flatten list-column into regular columns. We observe that for the same focal tree, we have information on all 20 competitor trees whose `dist` distance to the focal tree is $\leq$ `r comp_dist`: their unique tree-stem ID number, their species, and their basal area (in m$^2$) calculated as $\frac{\pi \times (\text{DBH/2})^2}{10000}$ where $DBH$ is the value from the earlier of the two censuses in cm. Saving our focal versus competitor information in list-column minimizes redundancy since we do not repeat information on the focal tree 20 times. 
+
+```{r, eval = FALSE}
+focal_vs_comp_scbi %>% 
+  filter(focal_ID == 4) %>% 
+  select(focal_ID, dbh, comp) %>% 
+  unnest(cols = "comp")
+```
+```{r scbi-focal-vs-comp-3, echo = FALSE}
+# Not sure why we need to do things this way:
+focal_vs_comp_scbi %>% 
+  filter(focal_ID == 4) %>% 
+  select(focal_ID, dbh, comp) %>% 
+  unnest(cols = "comp") %>% 
+  print(n = 5)
+```
+
+
+
+
+## Step 4: Fit model {#model-fit-predict}
+
+Now that we've identified all focal and corresponding competitor trees and saved this information in a data frame of type `focal_vs_comp`, the final step in our analysis sequence is to fit a model for the growth of all focal trees. Currently the `forestecology` package can only fit the competition Bayesian linear regression model outlined in Section \ref{competition-model} using the `comp_bayes_lm()` function. However, any model implemented in a function that similarly takes an input data frame of type `focal_vs_comp` as an argument can be easily swapped in. For our specific competition Bayesian linear regression model, we also specify prior distributions on all parameters of interest (here chosen to be the defaults as specified in `?comp_bayes_lm`). 
```

Comment:
Do yall think it's worth specifying how it is that one would wrap around a typical modeling function to use it with this package? Not sure if that's too much of a rabbit hole.

Reply:
Good point. I switched out "can easily be swapped in" in favor of "can be used"